crates/miden-tx/src/auth/tx_authenticator.rs

```diff
@@ -1,15 +1,55 @@
 use alloc::{collections::BTreeMap, string::ToString, sync::Arc, vec::Vec};
 
 use miden_objects::{
-    Felt, Word,
-    account::{AccountDelta, AuthSecretKey},
-    utils::sync::RwLock,
+    Felt, Hasher, Word, account::AuthSecretKey, crypto::SequentialCommit,
+    transaction::TransactionSummary, utils::sync::RwLock,
 };
 use rand::Rng;
 
 use super::signatures::get_falcon_signature;
 use crate::errors::AuthenticationError;
 
+// SIGNATURE DATA
+// ================================================================================================
+
+/// Data types on which a signature can be requested.
+///
+/// It supports three modes:
+/// - `TransactionSummary`: Structured transaction summary, recommended for authenticating
+///   transactions.
+/// - `Arbitrary`: Arbitrary payload provided by the application.
+/// - `Blind`: The underlying data is not meant to be displayed in a human-readable format. It can
+///   be assumed to already be a cryptographic commitment to some data.
```

Comment:
```suggestion
/// - `Blind`: The underlying data is not meant to be displayed in a human-readable format. It must
///   be a cryptographic commitment to some data.
```

Nit: Should we make the wording more strong and require it to be a commitment (given that it's not hashed in `to_commitment`?) I think for users constructing data for `SignatureData::Blind` it would clarify the requirements.

Reply:
Thanks, I definitely wasn't thinking in those terms - makes sense!